subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ivy/IvyDynamicRevisionRemoteResolveIntegrationTest.groovy

```diff
@@ -1012,7 +1012,7 @@ Required by:
         projectA2.jar.expectGet()
 
         then:
-        checkResolve "group:projectA:2.+": ["group:projectA:2.2", "didn't match versions 4.4, 3.0, 1.2, 1.1"]
+        checkResolve "group:projectA:2.+": ["group:projectA:2.2", "didn't match versions 3.0, 1.2, 1.1, 4.4"]
```

Comment:
Is there no way to have some of the performance gain while keeping the nicely sorted output?

Reply:
Not without changing the public API (and the internal ones) so that we have rich objects associated to reasons. Note that in this case, it's not sorted because we tried two different repositories. So the versions appear in the order they were checked, which, to some extent, is nice to know.